alg/gdaltransformer.cpp

```diff
@@ -595,18 +611,20 @@ GDALSuggestedWarpOutput2( GDALDatasetH hSrcDS,
 /*      Check if the computed target coordinates are revertable.        */
 /*      If not, try the detailed grid sampling.                         */
 /* -------------------------------------------------------------------- */
-    if( nFailedCount == 0 )
+    if( nFailedCount )
+    {
+        CPLDebug("WARP", "At least one point failed after direct transform");
+    }
+    else
     {
         memcpy(padfXRevert, padfX, nSamplePoints * sizeof(double));
         memcpy(padfYRevert, padfY, nSamplePoints * sizeof(double));
         memcpy(padfZRevert, padfZ, nSamplePoints * sizeof(double));
-        if( !pfnTransformer(pTransformArg, TRUE, nSamplePoints,
+        if( pfnTransformer(pTransformArg, TRUE, nSamplePoints,
                             padfXRevert, padfYRevert, padfZRevert, pabSuccess) )
         {
-            nFailedCount = 1;
-        }
-        else
-        {
+            double dfExpectedX = 0.0;
+            double dfExpectedY = 0.0;
```

Comment:
please revert those changes. best is to keep variables in the most nested possible scope